bounded/src/Data/Bounded/Text.hs

```diff
@@ -43,6 +43,9 @@ instance Show.Show BoundTextError where
     <> " but was "
     <> show actual
 
+instance Conversion Text BoundTextError where
+  convert = convert . show
```

Comment:
The goal of these instances is to make it possible to use a generic "parse" function for values that have partial conversion instances with these error types. We have a precedence of this elsewhere. 

Otherwise, at the principle level, I agree with you, which makes it evident that the past precedents are wrong and should be corrected. Essentially we should remove their text conversion instance and back-channel through `showc` to achieve the same goal as suggested.